### DISCUSSION TITLE
# Issue with running lighthouse on windows wsl2

### DISCUSSION THREAD

**Discussion by GuidoS** [GoogleChrome/lighthouse#13556] [Q&A]
I am trying to get lighthouse working on WSL2. I believe I have all the correct components installed. The issue I think is around the ip address of the WSL2 virtual machine. Is there a way to pass the local machine ip address?

Here is the setup I used and the code I ran.

## update wsl ubuntu 20 compponents

```sh
sudo apt update
sudo apt upgrade
lsb_release -a
```

> No LSB modules are available.  
> Distributor ID: Ubuntu  
> Description: Ubuntu 20.04.2 LTS  
> Release: 20.04  
> Codename: focal

```sh
sudo apt curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.37.2/install.sh | bash
nvm install --lts
nvm list
```

> default -> lts/\* (-> v14.16.0)  
> iojs -> N/A (default)  
> unstable -> N/A (default)  
> node -> stable (-> v14.16.0) (default)  
> stable -> 14.16 (-> v14.16.0) (default)  
> lts/\* -> lts/fermium (-> v14.16.0)  
> lts/argon -> v4.9.1 (-> N/A)  
> lts/boron -> v6.17.1 (-> N/A)  
> lts/carbon -> v8.17.0 (-> N/A)  
> lts/dubnium -> v10.24.0 (-> N/A)  
> lts/erbium -> v12.21.0 (-> N/A)  
> lts/fermium -> v14.16.0

```sh
npm install --save lighthouse chromium chrome-launch standard
```

## Code I am trying to run

```js
const fs = require("fs");
const lighthouse = require("lighthouse");
const chromeLauncher = require("chrome-launcher");

(async () => {
  const chrome = await chromeLauncher
    .launch({ chromeFlags: ["--headless"] })
    .catch((err) => {
      console.error(err);
      process.exit();
    });

  const options = {
    logLevel: "info",
    output: "html",
    onlyCategories: ["performance"],
    port: chrome.port,
  };
  const runnerResult = await lighthouse("https://example.com", options);

  // `.report` is the HTML report as a string
  const reportHtml = runnerResult.report;
  fs.writeFileSync("lhreport.html", reportHtml);

  // `.lhr` is the Lighthouse Result as a JS object
  console.log("Report is done for", runnerResult.lhr.finalUrl);
  console.log(
    "Performance score was",
    runnerResult.lhr.categories.performance.score * 100
  );

  await chrome.kill();
})();
```

## error message

> Error: connect ECONNREFUSED 127.0.0.1:41681
> at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16) {
> errno: -111,
> code: 'ECONNREFUSED',
> syscall: 'connect',
> address: '127.0.0.1',
> port: 41681
> }

**Comment by irfanfadilah** [comment]
I can confirm, pretty much the errors are same. In my case, I cannot use https://github.com/GoogleChromeLabs/bubblewrap in WSL2 because it needs to pass Lighthouse PWA checks.

**Comment by brendankenny** [comment]
> Is there a way to pass the local machine ip address?

the options passed to Lighthouse supports `hostname` as well as the `port` (which you're using in that snippet). If not included it defaults to `'localhost'`. Does specifying the `hostname` work for you?

**Comment by louisefindlay23** [comment]
I've tried specifying the hostname but that doesn't seem to take affect.

**Comment by creme332** [comment]
I was facing the same issue on WSL2 and resolved it by replacing `chrome-launcher` with `puppeteer` and then specifying `remote-debugging-port` = 9222.

First install the dependencies using `npm install lighthouse puppeteer` and then use the following code:

```js
import lighthouse from "lighthouse";
import puppeteer from "puppeteer";

(async () => {
  try {
    const browser = await puppeteer.launch({
      args: ["--remote-debugging-port=9222"],
    });

    const options = {
      logLevel: "info",
      output: "html",
      onlyCategories: ["performance"],
      port: 9222,
    };

    const url = "https://creme332.github.io/";
    const runnerResult = await lighthouse(url, options);

    console.log(
      "Performance score:",
      runnerResult.lhr.categories.performance.score * 100
    );

    await browser.close();
  } catch (err) {
    console.error(err);
  }
})();
```

Note: `puppeteer` installs its own Chrome called Chrome for Testing so you don't even need to install google chrome separately on WSL2.
